CONTRIBUTING.md

```diff
@@ -40,6 +40,12 @@ Code should adhere to the [coding style](#coding-style).
 Note that everybody follows these same guidelines.
 Core contributors, occasional contributors and maintainers are all alike.
 
+### Warnings 
+
+We use **Klocwork** as our static code analyzer. One of the most common issues that the Klocwork complaints about is comparing the return value of a function to nullptr instead of checking for non-zero return value.
+
+So to avoid creating more of this klocwork issue, please always prefer `if (!ptr)` instead of `if(ptr == nullptr)`.
```

Comment:
```suggestion
So to avoid creating more of these kinds of Klocwork issues, please always prefer `if (!ptr)` instead of `if(ptr == nullptr)`.
```
(nitpick)